### PR TITLE
Feature/async await

### LINKED ISSUE #584: Replace promises with async/await
The API side of the app makes heavy use of promises. Let's replace them with async/await for improved readability.

### PR DESCRIPTION
**Pull request by GregSithole**
Closes #584

- Added async-await to functions

### CODE REVIEW

**Review by jenkoian — COMMENTED** [review]
Thanks, it mostly looks really good but seems to break the gitlab integration. If you search on user `mwoolf` you can see the issue. Trying to dig in to find out more now.

File: api/controllers/pr/findPrs/gitlab/index.js [hunk 1]
```diff
@@ -4,139 +4,130 @@ const _ = require('lodash');
...
+      .filter((pr) => pr.repo_must_have_topic)
+      .map((pr) => pr.repo_id)
+  ).map((repo_id) => {
+    return gitlab.Projects.show(repo_id).then((res) => ({
+      repo_id,
+      topics: res.topics,
+    }));
+  });
+
+  const repoTopics = Promise.all(repoTopicRequests);
```
Comment (jenkoian) [inline api/controllers/pr/findPrs/gitlab/index.js:64]: This needs to have an `await` that should fix the gitlab bug 👍

Comment (GregSithole) [reply api/controllers/pr/findPrs/gitlab/index.js]: Sorry about that, but pushing up a fix now

**Comment by jenkoian** [comment]
Thanks @GregSithole

**Pull request merged**